protocol/superc20/mirror-factory-limited.md

```diff
@@ -0,0 +1,90 @@
+- `Sender Address`: In this case, it will be the Factory address itself, or the address of the factory's proxy. This works well, as it the factory can be deployed at the same address accross different chains. For chains in the superchain, it will be a predeploy. This predeploy, however, won't have a pretty address. Instead, it has to be deployed at the address where the L1 Factory is deployed so the address is maintained for all chains.
+- `Init Code`: The same proxy's init code must be used accross chains.
+
+The salt doesn't need to be more than the `OptimismMintableERC20Token` address. The `name` and `symbol`, and `decimals` can be fetched with the `token` address alone in the implementation. The factory will be the `Mirror`'s admin, and the proxy will have to either call `setStorage` immediately with the `token` at the time of deployment, or add a `TOKEN_SLOT` and update it on the constructor.
+
+The factory should be upgradable or have setters to allow the modification of the implementations for the current `Mirrors`. The deployment and update of already-deployed `Mirrors` to new implementations should be open to anyone.
+
+We can track versioning through numbers or the addresses of the implementations used.
+
+This design allows the `StandardBridge` or any contract that needs to check the legitimacy of a `Mirror` to call `MIRROR_FACTORY.getMirrorAddress()` and validate whether the calling user is a mirror that corresponds to a given `token`. As an illustrative example of a potential function of the `StandardBridge`:
+
+```solidity
+function mirrorMint(address _token, uint256 _amount) external {
+	address _mirror = MIRROR_FACTORY.getMirrorAddress(_token);
+	if (_mirror != msg.sender) revert();
+	///...mint logic
+}
+
+```
+
+Example code:
+
+```solidity
+contract MirrorFactory {
+	string public constant MIRROR_VERSION = "1.0.0";
+	Mirror public constant MIRROR_IMPL = <PLACEHOLDER_ADDRESS>;
+
+	function deployMirror(address _token) external returns (address) {
+		require(_isOptimismMintableERC20(_token), "not OptimismMintableERC20Token");
+
+		// Note: There's no _token slot in this proxy
+		Proxy _mirror = address(new L2ChugSplashProxy{salt: keccak256(abi.encode(_token))}(address(this), _token));
+
+		// Note: This function does not exist on L2ChugSplashProxy.
+		_mirror.setImplementation(MIRROR_IMPL);
+	}
+
+	function upgradeMirror(address _mirror) external {
+		// Note: This function does not exist on L2ChugSplashProxy.
+		_mirror.setImplementation(MIRROR_IMPL);
+	}
+
+	function _isOptimismMintableERC20(address _token) internal view returns (bool) {
+        return ERC165Checker.supportsInterface(_token, type(ILegacyMintableERC20).interfaceId)
+            || ERC165Checker.supportsInterface(_token, type(IOptimismMintableERC20).interfaceId);
+    }
+
+	function getMirrorAddress(address _token) external view returns (address _mirror) {
```

Comment:
Can we use a mapping that is populated rather than computing the address on chain? Otherwise we will need to forever pin the specific version of solc